### ISSUE TITLE
Dependencies

### ISSUE THREAD

**Issue by rejeep** [rejeep/prodigy.el#14]
Some of my services require other services in order to fully work. I suggest adding the property `:depends-on`. For example:

``` lisp
(prodigy-define-service
  :name "Script"
  :cwd "~/Code/script"
  :command "nodemon"
  :args '("app.coffee")
  :tags '(work))

(prodigy-define-service
  :name "Playground"
  :cwd "~/Code/playground"
  :command "nodemon"
  :args '("app.coffee")
  :tags '(work)
  :port 7171
  :depends-on '("Script"))
```

Starting service "Playground" would also start service "Script". It should not stop or restart dependencies since other services might rely on it.

**Comment by FrancisMurillo** [comment]
For me, I wanted to treat several services as one master service or set. Starting and stopping together instead of doing it individually. Not dependencies per se but a guarantee that several services work together instead.

I made an external module [prodigy-set.el](https://github.com/FrancisMurillo/prodigy-set.el) to do so. I thought about hacking with prodigy to get it done but there are concerns that might leak such as extra libraries (probably `deferred.el` but abstaining), overlapping or circular sets, action locking and others so I thought an external mode might offer better safety. 

My use case is with using a `tor` proxy with `w3m`. In order to join them, it needs a middleman proxy such as `polipo` and `privoxy` or both. I setup each proxy as a service but I want them to start in order when each is ready; likewise when one is down, I want them all down. The screencast shows my intention.

I do hope there is an official dependency management working to make things easier.

**Comment by rejeep** [comment]
The downside with groups is that some service might have to be started for another to be able to start. I think the elegancy with dependencies is that both issues are solved with it.

**Comment by leemeichin** [comment]
I've been looking at this and it seems like the current tags feature is close to it but doesn't quite do the thing we want. `:needs '(x y z)` would be enough to say 'ensure all of these are started before starting the current one'.

It's what docker-compose and CI pipelines do without overthinking it.
